test/test_jit.py

```diff
@@ -2833,6 +2833,34 @@ def foo(x, y):
         foo.graph.propagate_shapes((a, b), False)
         self.assertExpected(str(foo.graph))
 
+    def test_onnx_export_speculate(self):
+
+        class Foo(torch.jit.ScriptModule):
+            def __init__(self):
+                super(Foo, self).__init__()
+                self.linear = torch.jit.trace(torch.zeros(1, 10, dtype=torch.float))(nn.Linear(10, 20).float())
+
+            @torch.jit.script_method
+            def forward(self, x):
+                x += x
+                if True:
```

Comment:
How about make one of these levels of control a loop?

Reply:
y has to be assigned along all paths, so putting it into a loop would require putting `y = self.m(x)` before the loop but that would make it difficult to observe that the transpose call gets lifted.